R/match_pc_gpprac.R

```diff
@@ -0,0 +1,379 @@
+#' Recode Health Board code to 2018 standard
+#'
+#' @param hb_variable A vector of Health Board codes
+#'
+#' @return A vector of Health Board codes in the 2018 standard
+recode_health_boards <- function(hb_variable) {
+  hb_recoded <- dplyr::case_when(
+    # HB2014 to HB2018
+    hb_variable == "S08000018" ~ "S08000029",
+    hb_variable == "S08000027" ~ "S08000030",
+    # HB2019 to HB2018
+    hb_variable == "S08000031" ~ "S08000021",
+    hb_variable == "S08000032" ~ "S08000023",
+    TRUE ~ hb_variable
+  )
+  return(hb_recoded)
+}
+
+#' Recode HSCP code to 2018 standard
+#'
+#' @param hscp_variable A vector of HSCP codes
+#'
+#' @return A vector of HSCP codes in the 2018 standard
+recode_hscp <- function(hscp_variable) {
+  hscp_recoded <- dplyr::case_when(
+    # HSCP2016 to HSCP2018
+    hscp_variable == "S37000014" ~ "S37000032",
+    hscp_variable == "S37000023" ~ "S37000033",
+    # HSCP2019 to HSCP2018
+    hscp_variable == "S37000034" ~ "S37000015",
+    hscp_variable == "S37000035" ~ "S37000021",
+    TRUE ~ hscp_variable
+  )
+  return(hscp_recoded)
+}
+
+#' Rename existing geographies from temporary file to preserve them
+#' during transformations
+#'
+#' @param data A data frame
+#' @param vars A vector of variable names to rename. Default is that "lca",
+#' "HSCP", "DataZone" and "hbrescode" will be renamed
+#'
+#' @return A data frame with the variables from vars appended with "_old"
+rename_existing_geographies <- function(data,
+                                        vars = c("lca", "hscp2018", "datazone2011", "hbrescode")) {
```

Comment:
Can you explain the variables here? is the intention to use temp7 as the 'data'? I saw you had used the defaults as mentioned in the documentation but here they are now different